pkg/webhooks/machine_webhook.go

```diff
@@ -1381,6 +1381,14 @@ func validateAzureDataDisks(machineName string, spec *machinev1.AzureMachineProv
 			errs = append(errs, field.Invalid(fldPath.Child("lun"), disk.Lun, "each Data Disk must have a unique lun"))
 		}
 
+		switch disk.DeletionPolicy {
+		case machinev1.DiskDeletionPolicyTypeDelete, machinev1.DiskDeletionPolicyTypeDetach:
+			// Valid scenarios, do nothing
+		default:
+			errs = append(errs, field.Invalid(fldPath.Child("deletionPolicy"), disk.DeletionPolicy,
```

Comment:
I think there is a `field.Required` as well, is it worth having a distinction beteween an invalid value and not being set?

Reply:
Yes there is a `field.Required`, if you prefer I can add a separate case? Not sure what's the best practice here.

Reply:
I think it would be better to have a separate case for required vs invalid value, gives better feedback to the end user

Reply:
Ok, agree. Updated.